src/core_plugins/tagcloud/public/tag_cloud.js

```diff
@@ -116,116 +116,137 @@ class TagCloud extends EventEmitter {
     this._svgGroup.attr('height', this._size[1]);
   }
 
-  _processQueue() {
+  async _processPendingJob() {
 
-    if (!this._queue.length) {
-      this.emit('renderComplete');
+    if (!this._pendingJob) {
       return;
     }
 
-    if (this._inFlight) {
+    if (this._timeoutHandle || this._layoutIsUpdating || this._DOMisUpdating) {
       return;
     }
 
-    const job = this._queue.pop();
-    this._inFlight = true;
-
+    const job = await this._pickPendingJob();
     if (job.words.length) {
-      this._onLayoutEnd(job);
+      if (job.refreshLayout) {
+        await this._updateLayout(job);
+      }
+      await this._updateDOM(job);
+      const cloudBBox = this._svgGroup[0][0].getBBox();
+      this._cloudWidth = cloudBBox.width;
+      this._cloudHeight = cloudBBox.height;
+      this._allInViewBox = cloudBBox.x >= 0 &&
+        cloudBBox.y >= 0 &&
+        cloudBBox.x + cloudBBox.width <= this._element.offsetWidth &&
+        cloudBBox.y + cloudBBox.height <= this._element.offsetHeight;
     } else {
-      this._emptyCloud(job);
+      this._emptyDOM(job);
     }
+    this._currentJob = job;
 
+    if (this._pendingJob) {
+      this._processPendingJob();
+    } else {
+      this.emit('renderComplete');
+    }
+
+  }
+
+  async _pickPendingJob() {
+    return await new Promise((resolve) => {
+      this._timeoutHandle = setTimeout(async() => {
```

Comment:
thoughts on naming this something like pickJobtimeoutId?  it wasn't clear initially for me what it was keeping track of